robonomics_liability/src/robonomics_liability/player_service.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+#
+# Rosbag player service node.
+#
+
+from urllib.parse import urlparse
+from .LiabilityExecutionThread import LiabilityExecutionThread
+import rospy
+from robonomics_liability.srv import CreateRosbagPlayer, CreateRosbagPlayerRequest, CreateRosbagPlayerResponse
+from robonomics_liability.srv import StartRosbagPlayer, StartRosbagPlayerRequest, StartRosbagPlayerResponse
+from robonomics_liability.srv import StopRosbagPlayer, StopRosbagPlayerRequest, StopRosbagPlayerResponse
+from ipfs_common.msg import Multihash
+import ipfsapi
+import os
+import string
+import random
+
+
+class PlayerService:
+    def __init__(self):
+        rospy.init_node('player_service', anonymous=True)
+
+        ipfs_provider = urlparse(rospy.get_param('~ipfs_http_provider')).netloc.split(':')
+        self.ipfs_client = ipfsapi.connect(ipfs_provider[0], int(ipfs_provider[1]))
+
+        self.executions_work_directory = os.path.join(os.getcwd(), 'liabilities_executions')
+        self.master_check_interval = rospy.get_param('~master_check_interval')
+
+        self.recording_topics = list(filter(None, [x.strip() for x in rospy.get_param('~recording_topics').split(",")]))
+
+        self.player_threads = {}
+
+        def player_id_generator(multihash, namespace, size=8, chars=string.ascii_lowercase + string.digits):
+            id = ''.join(random.choice(chars) for _ in range(size))
+            return "{}{}_{}".format(multihash, namespace.replace('/', '_'), id)
+
+        def __new_player(msg):
+            ipfs_address = msg.ipfs_address
+            namespace = msg.namespace
+
+            response = CreateRosbagPlayerResponse()
+
+            if len(msg.namespace) < 2 or not msg.namespace.startswith('/'):
```

Comment:
http://docs.ros.org/melodic/api/rospy/html/rospy.names-module.html#valid_name could be better check